geopandas/geoseries.py

```diff
@@ -388,42 +383,21 @@ def to_crs(self, crs=None, epsg=None):
         epsg : int
             EPSG code specifying output projection.
         """
-        from fiona.crs import from_epsg
-
-        if crs is None and epsg is None:
-            raise TypeError("Must set either crs or epsg for output.")
-
         if self.crs is None:
             raise ValueError(
                 "Cannot transform naive geometries.  "
                 "Please set a crs on the object first."
             )
+        if crs is None and epsg is None:
+            raise TypeError("Must set either crs or epsg.")
 
-        if crs is None:
-            try:
-                crs = from_epsg(epsg)
-            except (TypeError, ValueError):
-                raise ValueError("Invalid epsg: {}".format(epsg))
-
-        # skip transformation if the input CRS and output CRS are the exact same
-        if _PYPROJ_VERSION >= LooseVersion("2.1.2") and pyproj.CRS.from_user_input(
-            self.crs
-        ).is_exact_same(pyproj.CRS.from_user_input(crs)):
+        # skip if the input CRS and output CRS are the exact same
+        crs = CRS.from_user_input(crs or epsg)
```

Comment:
I would pass the epsg explicitly to `CRS.from_epsg` (no need to allow people accidentally pass wrong input to `epsg=` IMO)

Reply:
Sounds like a good idea to me.